### PR TITLE
Add glob support for `inputPaths` in `TargetScript`

### PR DESCRIPTION
**Pull request by ibrahimoktay**
Hi Team,

It does not resolve any issue. See the slack thread [here](https://tuistapp.slack.com/archives/C018QG7U7SN/p1692195936248819)

### Motivation
I have some auto-generated files in my target and I want to be able to exclude them from the `inputPaths` in the script phase so that It can work properly, meaning that Xcode does not run the script all the time.

### Short description 📝

> This PR aims to add `excluding` capability for `inputPaths` by changing `inputPaths: [Path]` into  `inputPaths: [FileListGlob]` 
> There are some changes in the `ProjectDescription`. However, those changes were made to support backward compatibility. So, there should not be any need to migrate or adopt.

### How to test the changes locally 🧐

> * Check out the source branch 
> * Run the `TargetScriptManifestMapperTests`

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Tests/TuistLoaderTests/Models+ManifestMappers/TargetScript+ManifestMapperTests.swift [hunk 1]
```diff
@@ -33,24 +33,97 @@ final class TargetScriptManifestMapperTests: TuistUnitTestCase {
...
+            "foo/bar/c/c.swift",
+            "foo/bar/c/cTests.swift",
+        ])
+
+        let manifest = ProjectDescription.TargetScript.test(
+            name: "MyScript",
+            tool: "my_tool",
+            order: .pre,
+            arguments: ["arg1", "arg2"],
+            inputPaths: ["foo/bar/**/*.swift"],
```
Comment (ibrahimoktay) [inline Tests/TuistLoaderTests/Models+ManifestMappers/TargetScript+ManifestMapperTests.swift:51]: I was not planning to touch this part but It seems this test was checking the string equivalent. After changing it to a glob, I started resolving the glob earlier which means that it's actually looking for the folders/files. So, I followed the similar approach from `TargetTests`

File: Tests/TuistLoaderTests/Models+ManifestMappers/TargetScript+ManifestMapperTests.swift [hunk 2]
```diff
@@ -33,24 +33,97 @@ final class TargetScriptManifestMapperTests: TuistUnitTestCase {
...
+        let relativeSources = model.inputPaths.map { $0.relative(to: temporaryPath).pathString }
+
+        XCTAssertEqual(Set(relativeSources), Set([
+            "foo/bar/a.swift",
+            "foo/bar/b.swift",
+            "foo/bar/aTests.swift",
+            "foo/bar/bTests.swift",
+            "foo/bar/c/c.swift",
+            "foo/bar/c/cTests.swift",
+        ]))
```
Comment (pepicrft) [inline Tests/TuistLoaderTests/Models+ManifestMappers/TargetScript+ManifestMapperTests.swift:69]: What if we extract the list of paths into a variable to reuse it in `createFiles` and here?

**Review by pepicrft — APPROVED** [review]
This looks great @ibrahimoktay. I'm looping @tuist/core for additional feedback to make sure that we are not introducing any breaking changes with the modifications here. The only breaking change that I can think of is that users use the `Path` type directly instead of a `String`, in which case it'd cause the compilation of the manifest to fail.

File: Tests/TuistLoaderTests/Models+ManifestMappers/TargetScript+ManifestMapperTests.swift (hunk 2 above)
Comment (ibrahimoktay) [reply Tests/TuistLoaderTests/Models+ManifestMappers/TargetScript+ManifestMapperTests.swift:69]: I can do that but each list is unique. For example, the input list contains `.kt` file which I did not want to remove to be honest. It is another check when we do `*.swift`. So, it seemed a bit redundant to have another variable in this case.

**Comment by pepicrft** [comment]
@all-contributors add @ibrahimoktay for code

**Comment by ibrahimoktay** [comment]
@pepicrft Is there anything expected from me here ` run acceptance tests with Xcode (pull_request)` ? or anything I can help

**Review by danyf90 — APPROVED** [review]

**Pull request merged**